### PR TITLE
Fix BlockHeader versions and remove ConsensusManager from DB struct

### PR DESCRIPTION
**Pull request by mikethetike**
## Description

Removed a bunch of clones on ConsensusManager and consolidated the version code for Blockheader. 

## Motivation and Context

The main aim was to remove the ConsensusManager field from the BlockchainDatabase because it is only used in the constructor. Furthermore, it is only used for the blockchain version which I saw in most places was set to 0 or a value that didn't make sense.

I tried to apply this logic:
1. When creating a new block, use the value in the consensus rules.
2. When retrieving an old block, use the value it has

In most cases using the BlockBuilder,  the version would be set and then replaced immediately with the `header()` method. I introduced the `BlockHeader.into_builder()` method to use the header instead

## Types of changes

* [x] Bug fix (non-breaking change which fixes an issue)
* [ ] New feature (non-breaking change which adds functionality)
* [ ] Breaking change (fix or feature that would cause existing functionality to change)
* [x] Feature refactor (No new feature or functional changes, but performance or technical debt improvements)
* [ ] New Tests
* [ ] Documentation

## Checklist:

* [x] I'm merging against the `development` branch
* [x] I ran `cargo-fmt --all` before pushing
* [ ] My change requires a change to the documentation.
* [ ] I have updated the documentation accordingly.
* [ ] I have added tests to cover my changes.

### CODE REVIEW

File: base_layer/core/src/blocks/blockheader.rs [hunk 1]
```diff
@@ -160,6 +160,10 @@ impl BlockHeader {
         prev_pow.add_difficulty(&self.pow, self.achieved_difficulty());
         prev_pow.total_accumulated_difficulty()
     }
+
+    pub fn into_builder(self) -> BlockBuilder {
+        BlockBuilder::new(self.version)
```
Comment (SWvheerden) [inline base_layer/core/src/blocks/blockheader.rs]:
```suggestion
        BlockBuilder::new(self.version).with_header(self)
```
This should fix the tests

Comment (mikethetike) [reply base_layer/core/src/blocks/blockheader.rs]: Lol. My bad

**Comment by SWvheerden** [comment]
We can close this one. 
I rebased on this PR : #1472

**Comment by CjS77** [comment]
Changes covered in #1472

**Pull request closed without merging**